lib/convict.js

```diff
@@ -726,23 +726,21 @@ convict.addFormats = function(formats) {
 /**
  * Adds a new custom file parser
  */
-convict.addParser = function(parser) {
-  if (typeof parser.parse !== 'function') throw new Error('Parse function parser.parse must be a function');
+convict.addParser = function(parsers) {
+  if (!Array.isArray(parsers)) parsers = [parsers];
 
-  let formats = [];
-  formats = formats.concat(parser.formats || parser.format);
+  parsers.forEach(function(parser) {
+    if (!parser) throw new Error('Invalid parser');
+    if (!parser.extension) throw new Error('Missing parser extension');
+    if (!parser.parse) throw new Error('Missing parser parse function');
 
-  for (let i = 0; i < formats.length; i++) {
-    parsers[formats[i]] = parser.parse;
-  }
-};
+    if (typeof parser.parse !== 'function') throw new Error('Invalid parser.parse function');
 
-/**
- * Adds new custom file parsers
- */
-convict.addParsers = function(parsers) {
-  parsers.forEach(function(parser) {
-    convict.addParser(parser);
+    const extensions = !Array.isArray(parser.extension) ? [parser.extension] : parser.extension;
+    extensions.forEach(function(extension) {
+      if (typeof extension !== 'string') throw new Error('Invalid parser.extension');
```

Comment:
I think we should not care with type checks in this context. Adding a parser will fail if the type is not right. Adding type checks is worthy in JavaScript when a situation becomes critical or hard to debug. This is not the case here.